src/module.c

```diff
@@ -3891,7 +3891,7 @@ int RM_SetAbsExpire(RedisModuleKey *key, mstime_t expire) {
  * When async is set to true, db contents will be freed by a background thread. */
 void RM_ResetDataset(int restart_aof, int async) {
     if (restart_aof && server.aof_state != AOF_OFF) stopAppendOnly();
-    flushAllDataAndResetRDB(async? EMPTYDB_ASYNC: EMPTYDB_NO_FLAGS);
+    flushAllDataAndResetRDB((async? EMPTYDB_ASYNC: EMPTYDB_NO_FLAGS) | EMPTYDB_NOFUNCTIONS);
```

Comment:
i was paranoid thinking that maybe we did this on purpose and we don't recall why.
but i see the very first [commit](https://github.com/redis/redis/commit/52a49fda5ecec841a0658e11c2b131f7756fcb8e#diff-6109c354d7e009093f811238069b581bcb9bdbfc638d7d089814031776801632) in the original PR seemed to add EMPTYDB_WITHFUNCTIONS explicitly, so i assume there was no discussion that lead to deciding to flush functions in module.c